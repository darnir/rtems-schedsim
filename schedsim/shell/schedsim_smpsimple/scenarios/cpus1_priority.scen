#
#  Check increasing and decrease priority self with no change
#
rtems_init 1
task_create smp1 5
task_create smp2 8
dispatch
# GOAL: 0x0a010001 is heir and executing on CPU 0
cpus

task_priority smp1 4
# GOAL: 0x0a010001 is still heir and executing on CPU 0
cpus

task_priority smp1 7
# GOAL: 0x0a010001 is still heir and executing on CPU 0
cpus

task_priority smp1 8
# GOAL: 0x0a010002 is now heir and executing on CPU 0
cpus

task_priority smp1 5
# GOAL: 0x0a010001 is now heir and executing on CPU 0
cpus
