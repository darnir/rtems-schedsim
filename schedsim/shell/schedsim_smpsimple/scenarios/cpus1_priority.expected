Script File               : scenarios/cpus1_priority.scen
verbose                   : 1
==> 1: #
==> 2: #  Check increasing and decrease priority self with no change
==> 3: #
==> 4: rtems_init 1
  CPU 0: Thread Heir: 0x09010001 priority=255
  CPU 0: Thread Executing: 0x09010001 priority=255
==> 5: task_create smp1 5
Creating task with:  Preemption: yes Timeslicing: no
Task (smp1) created: id=0x0a010001, priority=5
Task (smp1) starting: id=0x0a010001, priority=5
  CPU 0: Thread Heir: 0x0a010001 priority=5
  CPU 0: Thread Executing: 0x0a010001 priority=5
==> 6: task_create smp2 8
Creating task with:  Preemption: yes Timeslicing: no
Task (smp2) created: id=0x0a010002, priority=8
Task (smp2) starting: id=0x0a010002, priority=8
==> 7: dispatch
==> 8: # GOAL: 0x0a010001 is heir and executing on CPU 0
==> 9: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
==> 10: 
==> 11: task_priority smp1 4
Task (0x0a010001) Change Priority from 5 to 4
==> 12: # GOAL: 0x0a010001 is still heir and executing on CPU 0
==> 13: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  4 / 0x0a010001 @  4          false
=== End of Ready Set of Threads
==> 14: 
==> 15: task_priority smp1 7
Task (0x0a010001) Change Priority from 4 to 7
==> 16: # GOAL: 0x0a010001 is still heir and executing on CPU 0
==> 17: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  7 / 0x0a010001 @  7          false
=== End of Ready Set of Threads
==> 18: 
==> 19: task_priority smp1 8
  CPU 0: Thread Heir: 0x0a010002 priority=8
  CPU 0: Thread Executing: 0x0a010002 priority=8
Task (0x0a010001) Change Priority from 7 to 8
==> 20: # GOAL: 0x0a010002 is now heir and executing on CPU 0
==> 21: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010002 @  8 / 0x0a010002 @  8          false
=== End of Ready Set of Threads
==> 22: 
==> 23: task_priority smp1 5
  CPU 0: Thread Heir: 0x0a010001 priority=5
  CPU 0: Thread Executing: 0x0a010001 priority=5
Task (0x0a010001) Change Priority from 8 to 5
==> 24: # GOAL: 0x0a010001 is now heir and executing on CPU 0
==> 25: cpus
=== CPU Status
          EXECUTING      /   HEIR             / SWITCH NEEDED
  CPU 0: 0x0a010001 @  5 / 0x0a010001 @  5          false
=== End of Ready Set of Threads
